src/main/scala/viper/gobra/frontend/Parser.scala

```diff
@@ -768,6 +769,25 @@ object Parser {
         ghostPrimaryExp |
         operand
 
+    // TODO: change delimiter to { and implement required ambiguity resolution, current: !< X, ..., Z !>
+    // declaredPred!<d1, ..., dn!>
+    lazy val fpredConstruct: Parser[PPredConstructor] =
+      (idnUse ~ predConstructArgs) ^^ { p => PPredConstructor(PFPredBase(p._1), p._2) }
+
+    lazy val mpredConstruct: Parser[PPredConstructor] =
+      (expression <~ ".") ~ idnUse ~ predConstructArgs ^^ {
+        case recv ~ id ~ args => PPredConstructor(PMPredBase(id, recv), args)
```

Comment:
```suggestion
        case recv ~ id ~ args => PPredConstructor(PMPredBase(id, recv).range(id, recv), args)
```
Same issue with position information as above